File: solutions/python/bob/1/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if hey_bob == "":
        return "Fine. Be that way!"

    if hey_bob.isupper():
        if hey_bob.endswith("?"):
            return "Calm down, I know what I'm doing!"
        else:
            return "Whoa, chill out!"

    if hey_bob.endswith("?"):
        return "Sure."

    return "Whatever."

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
    message = message.trim()

    let hasLetter : RegExpMatchArray | null = message.match(/[a-zA-Z]/)

    if (message.endsWith('?')) {
        if (hasLetter && message.toUpperCase() == message) {
            return 'Calm down, I know what I\'m doing!'
        } else {
            return 'Sure.'
        }
    }
    if (message.trim() == '') {
        return 'Fine. Be that way!'
    }
    if (hasLetter && message.toUpperCase() == message) {
        return 'Whoa, chill out!'
    }
    return 'Whatever.'
}
